tyrian/js/src/main/scala/tyrian/runtime/TyrianRuntime.scala

```diff
@@ -139,36 +138,51 @@ final class TyrianRuntime[F[_]: Async, Model, Msg](
   def toVNode(html: Html[Msg]): VNode =
     html match
       case Tag(name, attrs, children) =>
-        val as = js.Dictionary(
+        val as: List[(String, String)] =
           attrs.collect {
             case Attribute(n, v)   => (n, v)
             case NamedAttribute(n) => (n, "")
-          }: _*
-        )
-
-        val props =
-          js.Dictionary(attrs.collect { case Property(n, v) => (n, v) }: _*)
+          }
 
-        val events =
-          js.Dictionary(attrs.collect { case Event(n, msg) =>
-            (n, fun((e: dom.Event) => onMsg(msg.asInstanceOf[dom.Event => Msg](e))))
-          }: _*)
+        val props: List[(String, PropValue)] =
+          attrs.collect { case Property(n, v) => (n, v) }
 
-        val childrenElem: List[VNodeParam] =
-          children.map {
-            case t: Text            => VNodeParam.Text(t.value)
-            case subHtml: Html[Msg] => VNodeParam.Node(toVNode(subHtml))
+        val events: List[(String, dom.Event => Unit)] =
+          attrs.collect { case Event(n, msg) =>
+            (n, (e: dom.Event) => onMsg(msg.asInstanceOf[dom.Event => Msg](e)))
           }
```

Comment:
Ouch, that's really weird. I wonder if we haven't ported a test covering this yet, or it's just not covered by the snabbdom suite.

Reply:
Or I'm not using it right. 😅 
Mostly this is a blind port of the original implementation I inherited. I'm always prepared to accept that this might be me. 😉

Reply:
We are probably just missing the tests :) I'd guess they're covered by this suite which we have yet to port.
https://github.com/snabbdom/snabbdom/blob/fed2f31b8990258839e4547b967d3d9f21a94032/test/unit/eventlisteners.ts

Reply:
Hopefully should be fixed as of https://github.com/buntec/scala-js-snabbdom/pull/15.